### PR TITLE
feat: enable whitelisting multiple images

### PR DESCRIPTION
**Pull request by rishabh96b**
## Description

Enables multiple base images to be whitelisted. The image names can be set in the env variable using comma separated values. 
```dockerfile
ENV WHITELIST_IMAGE_NAME "alpine,scratch"
```

## Tests
With Dockerfile having single `FROM ..` line:
```log
[10/12/23: 14:54:31] [INFO] I'll do processing at [/home/rishabh/opstree/base-image-validator-proj/fork/BP-BASE-IMAGE-VALIDATOR//]
FROM alpine
RUN apk add --no-cache --upgrade bash
.
.
ENTRYPOINT [ "./build.sh" ]
[10/12/23: 14:54:36] [INFO] Got below lines in Dockerfile [FROM alpine]
[10/12/23: 14:54:36] [INFO] The base image is: alpine
alpine
[10/12/23: 14:54:36] [INFO] Image is whitelisted: alpine
[10/12/23: 14:54:36] [INFO] Congratulations BP_BASE_IMAGE_VALIDATOR succeeded!!!
```

With Dockerfile containting multiple `FROM..` lines or multi-stage Dockerfiles.
```
FROM golang:1.21
WORKDIR /src
..
FROM scratch
..
[10/12/23: 14:57:51] [INFO] Got below lines in Dockerfile [FROM golang:1.21
FROM scratch]
[10/12/23: 14:57:52] [INFO] Image is whitelisted: scratch
[10/12/23: 14:57:52] [INFO] Congratulations BP_BASE_IMAGE_VALIDATOR succeeded!!!
```
## Limitation
If a Dockerfile is using `AS` then this will not be successful.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -23,6 +22,22 @@ result=`getLineForAString ${DOCKERFILE_PATH} FROM`
 
 logInfoMessage "Got below lines in Dockerfile [${result}]"
 
-TASK_STATUS=`textExistsInALine "$result" "${WHITELIST_IMAGE_NAME}"`
+base_image=""
+# Consider the Dockerfiles with multi-stage builds.
+# In this case, the last one is considered as base image.
+while IFS= read -r line; do
```
Comment (sandy724) [inline build.sh]: This should go in a functoin

Comment (sandy724) [reply build.sh]: getBaseImage
Something like that

File: Dockerfile [hunk 2]
```diff
@@ -10,6 +10,6 @@ ENV SLEEP_DURATION 5s
 ENV ACTIVITY_SUB_TASK_CODE BP_BASE_IMAGE_VALIDATOR
 ENV VALIDATION_FAILURE_ACTION WARNING
 
-ENV WHITELIST_IMAGE_NAME alpine
+ENV WHITELIST_IMAGE_NAME "alpine,scratch"
```
Comment (sandy724) [inline Dockerfile]: Shall we change it to WHITELIST_IMAGES_NAME

**Review by sandy724 — CHANGES_REQUESTED** [review]

File: build.sh (hunk 1 above)
Comment (rishabh96b) [reply build.sh]: created a new utility file and added `getBaseImageFromFilteredDockerfile` function, updated in PR: https://github.com/OT-BUILDPIPER-MARKETPLACE/BP-BASE-SHELL-STEPS/pull/17/files
Will push changes respective to this repo also.

**Review by sandy724 — APPROVED** [review]
LGTM, assuming you have tested it

**Pull request merged**